.github/workflows/build-lint-test.yml

```diff
@@ -27,6 +27,7 @@ jobs:
           path: .yarn/cache
           key: yarn-cache-${{ runner.os }}-${{ steps.yarn-version.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
       - run: yarn install --immutable
+      - run: yarn dedupe --check
```

Comment:
The CI check might become annoying quickly, and it's not really an error. Git hook might be a better placement

Reply:
What is the scenario where you imagine it being annoying? 🤔

Reply:
Anytime I update a dependency my CI might become rejected, I now how to go back and check the logs, make a new commit, push and wait again.

Making this optional but automatic in a git hook would manage the size automatically and make for a nicer user experience

Reply:
Done!